app/api/routers/openneuro.py

```diff
@@ -1,105 +1,122 @@
-    # TODO: Handle network errors
-    gh_repo_url = f"https://github.com/OpenNeuroDatasets-JSONLD/{dataset_id}"
-    repo_url = (
-        f"https://api.github.com/repos/OpenNeuroDatasets-JSONLD/{dataset_id}"
+async def upload(
+    dataset_id: str,
+    data_dictionary: Annotated[UploadFile, File()],
+    changes_summary: Annotated[str, Form()],
+    name: Annotated[str, Form()],
+    email: Annotated[str, Form()],
+    affiliation: Annotated[str | None, Form()] = None,
+    # TODO: Do we need to check that it doesn't already start with a "@"? Or should we let the UI handle that?
+    gh_username: Annotated[str | None, Form()] = None,
+):
+    # TODO: Switch to using this Pydantic model directly for the /upload route form data once we
+    # upgrade the FastAPI version to >= 0.113.0 (and ensure that Pydantic v1 can still be used)
+    # See https://fastapi.tiangolo.com/tutorial/request-form-models/
+    contributor = Contributor(
+        name=name,
+        email=email,
+        affiliation=affiliation,
+        gh_username=gh_username,
+        changes_summary=utils.convert_literal_newlines(changes_summary),
     )
-    file_url = f"{repo_url}/contents/participants.json"
-    headers = {
-        "Accept": "application/vnd.github+json",
-        "Authorization": "Bearer " + GH_PAT,
-        "X-GitHub-Api-Version": "2022-11-28",
-    }
 
+    # TODO: Handle network errors
     upload_warnings = []
     file_exists = False
 
-    response = requests.get(repo_url, headers=headers)
-    # TODO: Should we explicitly handle 301 Moved permanently responses? These would fall under response.ok.
-    if not response.ok:
+    # Load private key from file to avoid newline issues when a multiline key is set in .env
+    with open(APP_PRIVATE_KEY_PATH, "r") as f:
+        APP_PRIVATE_KEY = f.read()
+
+    uploaded_file_contents = await data_dictionary.read()
+    try:
+        uploaded_dict = json.loads(uploaded_file_contents)
+    except json.JSONDecodeError:
         return JSONResponse(
             status_code=400,
             content=FailedUpload(
-                error=f"{response.status_code}: {response.reason}. Please ensure you have provided the correct repository ID."
+                error="The uploaded file is not a valid JSON file."
             ).dict(),
         )
 
-    current_file = requests.get(file_url, headers=headers)
-    if current_file.ok:
+    # Create a GitHub instance with the appropriate authentication
+    # (See https://pygithub.readthedocs.io/en/stable/examples/Authentication.html#app-installation-authentication)
+    auth = Auth.AppAuth(APP_ID, APP_PRIVATE_KEY)
+    gi = GithubIntegration(auth=auth)
+
+    # Get the installation ID for the Neurobagel Bot app (for the OpenNeuroDatasets-JSONLD organization)
+    installation = gi.get_org_installation(DATASETS_ORG)
+    installation_id = installation.id
+
+    g = gi.get_github_for_installation(installation_id)
```

Comment:
I think we do want to set up the authenticated instance in every request, because the access token generated for a given GH app installation (like we’re using here) will expire after a short period (1h by default I think?). This approach ensures that the token is refreshed as needed.